### DISCUSSION TITLE
Common voice notebook

### DISCUSSION THREAD

**Discussion by hammondm** [coqui-ai/STT#2054] [Q&A] [answered]
all

As far as I can tell the notebook that exemplifies training with common voice doesn't actually train.

It's set to run only one epoch, but if you look at the status updates, it reports no loss for training. If you increase the number of epochs, the training loss is a constant 0 and the validation loss is 300+, but never changes.

I've confirmed this behavior on colab and on my own machine.

mike h.

**Comment by JRMeyer** [comment, marked as answer]
Thanks for opening the discussion @hammondm 👍

I'll take a look and hopefully we can figure this out

**Reply by JRMeyer** [reply]
@hammondm -- the notebook Common Voice notebook `train_with_common_voice.ipynb` works fine for me on colab:

```
I Initializing all variables.
I STARTING Optimization
Epoch 0 |   Training | Elapsed Time: 0:00:22 | Steps: 88 | Loss: 62.146426     
Epoch 0 | Validation | Elapsed Time: 0:00:06 | Steps: 77 | Loss: 48.618259 | Dataset: serbian/sr-data/dev.csv
I Saved new best validating model with loss 48.618259 to: serbian/checkpoints/best_dev-88
--------------------------------------------------------------------------------
I FINISHED optimization in 0:00:29.284440
```

I think this discussion sounds more like a technical issue, though. Can you open an issue with exact steps to replicate here https://github.com/coqui-ai/STT/issues

**Reply by hammondm** [reply]
jm

I figured it out. The notebook says you can uncomment the batchsize lines in the config step if you have a GPU. I do, so I did.

That seems to be the problem. If I comment those out again, training works.

I would imagine the problem is that the batch sizes given there are bigger than the size of the actual data!

mh
